applications/OptimizationApplication/python_scripts/utilities/opt_convergence.py

```diff
@@ -22,6 +22,26 @@ def CreateConvergenceCriteria(parameters: Kratos.Parameters, optimization_proble
         raise RuntimeError(f"CreateConvergenceCriteria: unsupported convergence type {type}.")
 
 class MaxIterConvCriterion:
+    """
+    A convergence criterion class that checks if the maximum number of iterations has been reached.
+
+    Methods
+    -------
+    GetDefaultParameters():
+        Returns the default parameters for the convergence criterion.
+    
+    __init__(parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+        Initializes the convergence criterion with the given parameters and optimization problem.
+    
+    IsMaxIterationsReached():
+        Checks if the maximum number of iterations has been reached.
+    
+    IsConverged(search_direction=None) -> bool:
+        Checks if the optimization problem has converged based on the maximum number of iterations.
+    
+    GetInfo() -> dict:
+        Returns a dictionary with information about the current state of convergence.
+    """
```

Comment:
@matekelemen, does this way of writing method docs within the class doc works? can doxygen decipher this and put it to correct method?